.github/workflows/check-branch.yml

```diff
@@ -0,0 +1,72 @@
+name: Check branch version
+
+# This is a separate workflow from the main build-and-test workflow
+# because we want it to run when a PR's target branch changes, and
+# it'd be prohibitively to re-run the main workflow every time that
```

Comment:
`# it'd be prohibitively to re-run` -> `# prohibitively expensive/?`

Reply:
Ah, I meant to include "time-consuming"

Reply:
fixed